src/actor/editor.rs

```diff
@@ -0,0 +1,111 @@
+            world_sender,
+        }
+    }
+
+    pub async fn run(mut self) {
+        self.editor_websocket_conn
+            .send(Message::Text(
+                serde_json::to_string(&ControlPlaneResponse::SyncEditorChanges(())).unwrap(),
+            ))
+            .await
+            .unwrap();
+        loop {
+            tokio::select! {
+                Some(msg) = self.mailbox.recv() => {
+                    debug!("EditorActor: received message from mailbox: {:?}", msg);
+                    match msg {
+                        EditorActorRequest::DocToSrcJump(jump_info) => {
+                            let Ok(_) = self.editor_websocket_conn.send(Message::Text(
+                                serde_json::to_string(&ControlPlaneResponse::EditorScrollTo(jump_info)).unwrap(),
+                            )).await else {
+                                warn!("EditorActor: failed to send DocToSrcJump message to editor");
+                                break;
+                            };
+                        }
+                    }
+                }
+                Some(Ok(Message::Text(msg))) = self.editor_websocket_conn.next() => {
+                    let Ok(msg) = serde_json::from_str::<ControlPlaneMessage>(&msg) else {
+                        warn!("failed to parse jump request: {:?}", msg);
+                        continue;
+                    };
+                    match msg {
+                        ControlPlaneMessage::SrcToDocJump(jump_info) => {
+                            debug!("EditorActor: received message from editor: {:?}", jump_info);
+                            self.world_sender.send(WorldActorRequest::SrcToDocJumpResolve(jump_info))
+                        }
+                        ControlPlaneMessage::SyncMemoryFiles(memory_files) => {
+                            debug!("EditorActor: received message from editor: SyncMemoryFiles {:?}", memory_files.files.keys().collect::<Vec<_>>());
+                            self.world_sender.send(WorldActorRequest::SyncMemoryFiles(memory_files))
+                        }
+                        ControlPlaneMessage::UpdateMemoryFiles(memory_files) => {
+                            debug!("EditorActor: received message from editor: UpdateMemoryFiles {:?}", memory_files.files.keys().collect::<Vec<_>>());
+                            self.world_sender.send(WorldActorRequest::UpdateMemoryFiles(memory_files))
+                        }
+                        ControlPlaneMessage::RemoveMemoryFiles(memory_files) => {
+                            debug!("EditorActor: received message from editor: RemoveMemoryFiles {:?}", &memory_files.files);
+                            self.world_sender.send(WorldActorRequest::RemoveMemoryFiles(memory_files))
+                        }
+                    }.unwrap();
+                }
+            }
+        }
+        info!("EditorActor: ws disconnected, shutting down whole program");
+        std::process::exit(0);
```

Comment:
when editor disconnects, it shutdown itself. this should effectively fix #81 . Although works in vscode extension code are left untouched now